docs/source/serving/high-availability.rst

```diff
@@ -0,0 +1,207 @@
+.. _skyserve-high-availability-controller:
+
+=========================================
+High Availability SkyServe Controller
+=========================================
+
+Overview
+--------
+By default, the SkyServe controller runs as a single instance (either a VM or a Kubernetes Pod). If this instance fails due to node issues, pod crashes, or other unexpected events, the service endpoint becomes unavailable until the controller is manually recovered or relaunched.
+
+To enhance resilience and ensure service continuity, SkyServe offers a High Availability mode for its controller. When enabled, the controller leverages the automatic recovery feature of Kubernetes Deployments to automatically recover from failures.
+
+Benefits of high availability controller:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* **Automatic Recovery:** The controller and load balancer can automatically restart after crashes or node failures.
+* **Service Continuity:** Minimizes downtime for your served applications.
+* **Managed by Kubernetes:** Relies on robust, industry-standard Kubernetes features for state management and process supervision.
+
+Prerequisites
+-------------
+* **Kubernetes Cluster:** High Availability mode is **currently only supported when using Kubernetes** as the cloud provider for the SkyServe controller. You must have a Kubernetes cluster configured for SkyPilot. See :ref:`Kubernetes Setup <kubernetes-setup>` for details.
+* **Persistent Kubernetes:** The underlying Kubernetes cluster (control plane and nodes) must be running persistently. If using a local Kubernetes deployment (e.g., Minikube, Kind via ``sky local up``), the machine hosting the cluster must remain online.
+
+.. note::
+    Currently, High Availability mode is only supported for Kubernetes. Support for other clouds (e.g., AWS, GCP, Azure VMs) is under development.
+
+How to enable high availability mode
+-------------------------------------
+To enable High Availability for the SkyServe controller, set the ``high_availability`` flag to ``true`` within the ``serve.controller`` section of your :ref:`SkyPilot configuration <config-yaml>`:
```

Comment:
How about we lower case `High Availability` everywhere?

Reply:
Good point! Let's use lower cases for consistency.